src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -107,15 +103,46 @@ private BindingDataResolver GetResolver(PropertyInfo propInfo, INameResolver nam
                 return GetAppSettingResolver((string)originalValue, appSettingAttr, nameResolver, propInfo);
             }
             // try to resolve with auto resolve ({...}, %...%)
-            if (autoResolveAttr != null && originalValue != null)
+            if (autoResolveAttr != null)
             {
-                _autoResolves[propInfo] = autoResolveAttr;
-                return GetTemplateResolver((string)originalValue, autoResolveAttr, nameResolver, propInfo, contract);
+                if (propInfo.PropertyType != typeof(string))
+                {
+                    throw new InvalidOperationException($"Property '{propInfo.Name}' with AutoResolve must be type string.");
+                }
+                var str = (string)originalValue;
```

Comment:
The string requirement is only if you have a [AutoResolve] on it, so I still need the call on line 91. 

---
In reply to: [116878127](https://github.com/Azure/azure-webjobs-sdk/pull/1148#discussion_r116878127) [](ancestors = 116878127)